File: Java/dependency-injection/conforming-container-antipattern/src/main/java/com/github/grzesiek_galezowski/ConformingContainerAntipattern/InMessages/StartMessage.java
package com.github.grzesiek_galezowski.ConformingContainerAntipattern.InMessages;

import com.github.grzesiek_galezowski.ConformingContainerAntipattern.Interfaces.InboundMessage;
import com.github.grzesiek_galezowski.ConformingContainerAntipattern.Interfaces.DataDestination;
import com.github.grzesiek_galezowski.ConformingContainerAntipattern.Services.Authorization;

public class StartMessage implements InboundMessage {
  public void authorizeUsing(Authorization authorizationRules) {
    System.out.println("Authorizing start with " + authorizationRules);
  }

  public void writeTo(DataDestination dataDestination) {
    dataDestination.add("Writing start to " + dataDestination);
  }
}

File: CSharp/DependencyInjection/DependencyInjection/_1_ControlFreak/InMessages/StartMessage.cs
using System;
using DependencyInjectionBefore._1_ControlFreak.Interfaces;
using DependencyInjectionBefore._1_ControlFreak.Services;

namespace DependencyInjectionBefore._1_ControlFreak.InMessages
{
  class StartMessage : AcmeMessage
  {
    public void AuthorizeUsing(ActiveDirectoryBasedAuthorization authorizationRules)
    {
      Console.WriteLine("Authorizing Start with " + authorizationRules);
    }

    public void WriteTo(DataDestination dataDestination)
    {
      dataDestination.Add("Writing Start to " + dataDestination);
    }
  }
}